build.sbt

```diff
@@ -2,30 +2,85 @@ name := "delphi"
 version := "1.0.0-SNAPSHOT"
 scalaVersion := "2.12.4"
 
+import sys.process._
+import java.io.File
 
 
+lazy val repos = List("delphi-webapi", "delphi-crawler", "delphi-cli", "delphi-management")
+lazy val delphiRepos = "repos"
+
+def cloneAll = Command.command("clone-all") {
+  state =>
+    Process(s"mkdir -p $delphiRepos").!
+    val currDir = System.getProperty("user.dir")
+    for (repo <- repos) {
+      val bash = "/bin/bash"
+      val op = "-c"
+      val clone = s"git clone https://github.com/delphi-hub/$repo"
+      Process(Seq(bash, op, clone), new File(currDir + "/" + delphiRepos)).!
+    }
+    state
+}
+
+
+
+
+
+def delete = Command.command("delete") {
+  state =>
+    val currDir = System.getProperty("user.dir")
+    val cmd = "rm -rf " + currDir + "/" + delphiRepos
+    Process(cmd).!
+    state
+}
+
+
+lazy val currentBranch = taskKey[String]("Get current git branch")
+
+currentBranch := {
+  Process("git rev-parse --abbrev-ref HEAD").!!
+}
+
+
+lazy val pull = taskKey[Unit]("Pull Changes")
+
+pull := {
+  val branch = currentBranch.value
+  for (repo <- repos) {
+    BuildUtils.pull(repo, branch)
+  }
+
+}
+
+Compile / compile := ((Compile / compile) dependsOn pull).value
+
+commands += cloneAll
+commands += delete
 lazy val root = (project in file("."))
-                .aggregate(cli, crawler, management, webapi, webapp)
+  .aggregate(crawler,webapi,cli,management,webapp)
```

Comment:
`repos` is directory containing all git components. It is added `.gitignore`. `clone-all` command clones the repositories/